### PR TITLE
Script only "quick start" mode

### PR DESCRIPTION
**Pull request by nick-l-o3de**
## What does this PR do?
This is the first step of "script only" Quick start no-c++-compile feature, as per the RFC in SIG-core

I have tested it in Linux, ubuntu 23.x and windows 11.

Tested under linux by using the Project Manager to instantiate a Script Only Quick Start Project and see that it can get into the editor without any compilation  or 3p package getting except for python (which is going to be important to have present anyway).

Tested under Windows using the same method.

To test, you'd have to build an installer out of this, and then use that installer to make Script only Project. (it will not work in code mode).

Of note:
The template files and icons are identical to the default project, the only files of note that are different in the template are the template itself and the engine finder json.

### CODE REVIEW

**Comment by nick-l-o3de** [comment]
preview
 the horrible Icon I made.

File: CMakeLists.txt [hunk 1]
```diff
@@ -55,6 +55,18 @@ include(cmake/O3DEJson.cmake)
 include(cmake/Subdirectories.cmake)
 include(cmake/TestImpactFramework/LYTestImpactFramework.cmake) # Put at end as nothing else depends on it
 
+get_property(o3de_script_only GLOBAL PROPERTY "o3de_script_only")
+if (o3de_script_only)
+    if (NOT INSTALLED_ENGINE)
```
Comment (lumberyard-employee-dm) [inline CMakeLists.txt:60]: These two conditions could be checked in the same if statement.
```suggestion
if (o3de_script_only AND NOT INSTALLED_ENGINE)
```

File: Code/Framework/AzNetworking/AzNetworking/Utilities/EncryptionCommon.cpp [hunk 2]
```diff
@@ -537,7 +537,13 @@ namespace AzNetworking
         SSL_CTX_set_cookie_verify_cb(context, VerifyCookieCallback);
 
         // Automatically generate parameters for elliptic-curve diffie-hellman (i.e. curve type and coefficients).
+
+        // note that the below generates a warning depending on your verison of openssl - on some, these functions
```
Comment (lumberyard-employee-dm) [inline Code/Framework/AzNetworking/AzNetworking/Utilities/EncryptionCommon.cpp:541]: typo: verison -> version
```suggestion
        // note that the below generates a warning depending on your version of openssl - on some, these functions
```

File: Templates/ScriptOnlyProject/Template/cmake/EngineFinder.cmake [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    message(VERBOSE "No compatible engine found from CMAKE_MODULE_PATH '${CMAKE_MODULE_PATH}'.")
+endif()
+
+# before proceeding, check and set a global variable if its a "script only" project
+file(READ ${CMAKE_CURRENT_SOURCE_DIR}/project.json o3de_project_json)
+string(JSON project_is_script_only ERROR_VARIABLE json_error GET ${o3de_project_json} "script_only")
+if(project_is_script_only AND NOT json_error)
+    # we only set it if there's no error and it is a script only project
+    message(STATUS "This is a script-only project, no C++ compiler will be used")
+    set_property(GLOBAL PROPERTY "o3de_script_only" TRUE)
```
Comment (lumberyard-employee-dm) [inline Templates/ScriptOnlyProject/Template/cmake/EngineFinder.cmake:42]: Normally for global properties, we have been setting them using a naming scheme of all caps.
Such as "O3DE_SCRIPT_ONLY".

**Review by phuchau1989 — APPROVED** [review]

**Comment by nick-l-o3de** [comment]
So the next thing to do is to figure out, is this good enough, or should I do more before taking it out of draft?
Its immediately useful...
but you can't export a project or use a standalone launcher, only use the editor

File: engine.json [hunk 4]
```diff
@@ -129,7 +129,8 @@
         "Templates/PythonToolGem",
         "Templates/RemoteRepo",
         "Templates/ScriptCanvasNode",
-        "Templates/UnifiedMultiplayerGem"
+        "Templates/UnifiedMultiplayerGem",
+        "Templates/ScriptOnlyProject"
```
Comment (lumberyard-employee-dm) [inline engine.json:133]: The templates list are kept in alphabetical order.
So "ScriptOnlyProject" should be above "UnifiedMultiplayerGem"
```suggestion
        "Templates/ScriptOnlyProject",
        "Templates/UnifiedMultiplayerGem"
```

**Comment by lumberyard-employee-dm** [comment]
> So the next thing to do is to figure out, is this good enough, or should I do more before taking it out of draft? Its immediately useful... but you can't export a project or use a standalone launcher, only use the editor

I think this good enough to take it out of a Draft PR and has use already for Linux and Windows.
In theory it can work for MacOS as well since we aren't building any Bundles using the O3DE SDK, but only using existing ones that have already been built.

**Comment by nick-l-o3de** [comment]
I won't be able to test it on MacOS, and haven't tested it on windows (I run on linux).   I'll try to at least get a windows machine compiling it before I switch it from draft to actual.

However, there are some powerful caveats for this thing, for example, inability to make an actual shipping game out of it using any kind of build-and-pack script.  There is no launcher.  

Maybe it would be worth figuring out a way to communicate these caveats?  But how

**Comment by nick-l-o3de** [comment]
I tested this on windows, and it does appear to work there too.

**Review by lumberyard-employee-dm — APPROVED** [review]
This is good with me then.

This should allow users to go from install into the Editor a lot quicker(though the AP still needs to process all the critical assets before the Editor is usable)

**Review by AMZN-alexpete — APPROVED** [review]

**Pull request merged**